io.sloeber.ui/src/io/sloeber/ui/monitor/views/SerialMonitor.java

```diff
@@ -135,10 +136,11 @@
 
 	private static final String MY_FLAG_MONITOR = "FmStatus"; //$NON-NLS-1$
 	static final String uri = "h tt p://ba eye ns. i t/ec li pse/d ow nlo ad/mo nito rSta rt.ht m l?m="; //$NON-NLS-1$
+	static final String NL = System.getProperty("line.separator");
 
 	private static SerialMonitor instance = null;
 
-	public static SerialMonitor getSerialMonitor() {
+	public static synchronized SerialMonitor getSerialMonitor() {
```

Comment:
I assume this means the comment " Must be run from the ui thread" can be removed from derialconnection.add and serial connection.remove

Reply:
I just did that because I figured any singleton's get... method should be sync'd.

Reply:
I'm not sure what a singleton is. I just want to keep the doc up to date as possible

Reply:
SerialMonitor implements the singleton pattern in a textbook fashion, other than not protecting the getSerialMonitor method. You have a single instance held in a private static field, and you get that instance via the getSerialMonitor method.

The normal practice is to protect against two threads running getXxx at the same time by synchronizing it. Then any number of threads can call simultaneously it and you'll still only ever have one instance created because only one thread can hit the == null test in the method at a time.

So that comment probably isn't necessary if the method is sync'd, as long as there is no other reason it should only be called from the UI thread.

Reply:
OK, not quite textbook singleton. The constructor should be private, but it can't be because Eclipse wants to call it via introspection.

But still, the idea is the same. You only want one of them.

Actually, if Eclipse is creating the instance via the constructor, does the null check in getSerialMonitor ever trigger?